src/internal/connector/onedrive/permission.go

```diff
@@ -253,11 +295,74 @@ func RestorePermissions(
 
 		pbody.SetRecipients([]models.DriveRecipientable{rec})
 
-		_, err := service.Client().DrivesById(driveID).ItemsById(itemID).Invite().Post(ctx, pbody, nil)
+		np, err := service.Client().DrivesById(driveID).ItemsById(itemID).Invite().Post(ctx, pbody, nil)
 		if err != nil {
 			return graph.Wrap(ctx, err, "setting permissions")
 		}
+
+		if usePermissionIDMappings {
+			permissionIDMappings[p.ID] = ptr.Val(np.GetValue()[0].GetId())
+		}
 	}
 
 	return nil
 }
+
+// RestorePermissions takes in the permissions that were added and the
+// removed(ones present in parent but not in child) and adds/removes
+// the necessary permissions on onedrive objects.
```

Comment:
Comment could use revision as now we're just taking in the item metadata and computing the added/removed permission sets